pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -221,35 +210,53 @@ func (c *Cluster) removeExtraMgrs(daemonIDs []string) {
 	}
 }
 
-// ReconcileActiveMgrServices reconciles the services if the active mgr is the one running
-// in the sidecar
-func (c *Cluster) ReconcileActiveMgrServices(daemonNameToUpdate string) error {
-	// If the services are already set to this daemon, no need to attempt to update
-	svc, err := c.context.Clientset.CoreV1().Services(c.clusterInfo.Namespace).Get(c.clusterInfo.Context, AppName, metav1.GetOptions{})
+// UpdateActiveMgrLabel updates the mgr_role label value to either
+// active or standby depending on the status of the ceph mgr running
+// in the pod
+func (c *Cluster) UpdateActiveMgrLabel(daemonNameToUpdate string) error {
+
+	mgrName := fmt.Sprintf("%s-%s", AppName, daemonNameToUpdate)
+	logger.Infof("Upadting active mgr label for daemon %s", mgrName)
+	deployment, err := c.context.Clientset.AppsV1().Deployments(c.clusterInfo.Namespace).Get(c.clusterInfo.Context, mgrName, metav1.GetOptions{})
 	if err != nil {
-		logger.Errorf("failed to check current mgr service, proceeding to update. %v", err)
-	} else {
-		currentDaemon := svc.Spec.Selector[controller.DaemonIDLabel]
-		if currentDaemon == daemonNameToUpdate {
-			logger.Infof("mgr services already set to daemon %q, no need to update", daemonNameToUpdate)
-			return nil
-		}
-		logger.Infof("mgr service currently set to %q, checking if need to update to %q", currentDaemon, daemonNameToUpdate)
+		logger.Infof("Cannot get deployment with name %s", mgrName)
+		return err
 	}
 
-	activeName, err := c.getActiveMgr()
+	// Get the pods of the deployment
+	pods, err := c.context.Clientset.CoreV1().Pods(c.clusterInfo.Namespace).List(c.clusterInfo.Context, metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(deployment.Spec.Selector),
+	})
 	if err != nil {
+		logger.Infof("Cannot get pod with name %s", deployment.Name)
 		return err
 	}
-	if activeName == "" {
-		return errors.New("active mgr not found")
+
+	activeMgr, err := c.getActiveMgr()
+	if err != nil || activeMgr == "" {
+		logger.Infof("cannot update active mgr, no active mgr found. err=%v", err)
+		return err
 	}
-	if daemonNameToUpdate != activeName {
-		logger.Infof("no need for the mgr update since the active mgr is %q, rather than the local mgr %q", activeName, daemonNameToUpdate)
-		return nil
+
+	for i, pod := range pods.Items {
+		labels := pod.GetLabels()
+		cephDaemonId := labels[controller.DaemonIDLabel]
+
+		if activeMgr == cephDaemonId {
+			labels[mgrRoleLabelName] = "active"
```

Comment:
What if the pod already has the correct active/standby label? Then we can skip the update below.

Reply:
Fixed